src/GraphQL.Federation/Instrumentation/HttpRequestExtensions.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using System.Linq;
+using Microsoft.AspNetCore.Http;
+
+namespace GraphQL.Federation.Instrumentation
+{
+    /// <summary>
+    /// <see cref="HttpRequest"> extension methods for checking headers.
+    /// </summary>
+    public static class HttpRequestExtensions
```

Comment:
It will be better to move all transport-releted stuff into some other place, for example, into [server](https://github.com/graphql-dotnet/server) project. GraphQL.NET is transport agnostic.